cli/src/pcluster/config/validators.py

```diff
@@ -1125,11 +1124,10 @@ def architecture_os_validator(param_key, param_value, pcluster_config):
     warnings = []
 
     architecture = pcluster_config.get_section("cluster").get_param_value("architecture")
-    allowed_oses = get_supported_os_for_architecture(architecture)
-    if param_value not in allowed_oses:
+    if param_value not in SUPPORTED_OSS:
```

Comment:
This is a redundant logic that is already covered by `"allowed_values": SUPPORTED_OSS,` in `mappings.py`. I would remove this validator normally, but I know we would need to add a warning for centos7+arm in this validator later.

Can we either add a `TO-DO` comment in the code or a placeholder warning message to track why we are still keeping this validator for now, and make sure we can revisit this code later

Reply:
Add validation message and refactored run_architecture_validator_test()